### ISSUE TITLE
VSCode Semgrep Extension failing to update rules behind inspecting proxy

### ISSUE THREAD

**Issue by mitch-athena** [semgrep/semgrep-vscode#157]
I'm behind an inspecting proxy and having this issue running semgrep in VSCode and already have the `REQUESTS_CA_BUNDLE` exported, as well as `CERT_PATH`, `CERT_DIR`, `SSL_CERT_FILE`, `SSL_CERT_DIR` as this also fixes TLS issues for PIP. Any idea what else is needed to get this working?

Error Notification in VSCode:
```
Fatal error: Failed to download config from https://semgrep.dev/c/p/default: TLS failure: authentication failure: invalid certificate chain
```

Error in output log from Semgrep extension in VSCode:
```log
Refreshing rules
[73.00][ERROR]: Uncaught async exception: Fatal error: Failed to download config from https://semgrep.dev/c/p/default: TLS failure: authentication failure: invalid certificate chain
```

No issue running semgrep from the CLI though.

_Originally posted by @mitch-athena in https://github.com/semgrep/semgrep/issues/5084#issuecomment-2244055360_

I noticed this has been raised before here https://github.com/semgrep/semgrep-vscode/issues/62 and was supposedly fixed in https://github.com/semgrep/semgrep-vscode/pull/63, however the issue seems to persist.

**Comment by ajbt200128** [comment]
Hey so it turns out that issue was never actually fixed, and we have been working on a fix for awhile now. It'll probably be another month until we can get that fix through though as it depends on updating a dependency.

**Comment by mitch-athena** [comment]
Glad it wasn't just me then 😅 

Will check back here later then, thanks!

**Comment by ajbt200128** [comment]
Not yet, there are some PRs that need to be merged into the libraries we use before this is possible

**Comment by vbakke** [comment]
I can confirm that this issue applies to the Semgrep plugin in IntelliJ as well.

`Semgrep: Fatal error: Failed to download config from https://semgrep.dev/c/p/default, returned code 0: Error: self-signed certificate in certificate chain at TLSSocket.onConnectSecure (node:_tls_wrap:1659:34) at TLSSocket.emit (node:events:517:28) at TLSSocket._finishInit (node:_tls_wrap:1070:8) at ssl.onhandshakedone (node:_tls_wrap:856:12)`

**Comment by manumns** [comment]
I also get this error.
In my case, it also fails to login to Sempgrep in the first place when I select `>Semgrep: Sign in`

```
Failed to complete login process: ERROR Login Failed! We had an unexpected failure while trying to make a network request to https://semgrep.dev/api/agent/tokens/requests: TLS failure: authentication failure: invalid certificate chain Please check network settings and try again or reach out to Semgrep support at [36m[4mhttps://semgrep.dev/docs/support/[39m[24m
```
I also have no issue running semgrep from the CLI.

**Comment by vbakke** [comment]
Maybe https://gitlab.com/alelec/pip-system-certs contains any hints for a potential solution?

**Comment by cerojasg1** [comment]
It seems the issue is still happening, doesn't it? 
![image](https://github.com/user-attachments/assets/5bf272c3-1b68-4200-91d2-81af52ed7377)

**Comment by ajbt200128** [comment]
yup this is still an issue. Should be fixed once some updates to some networking related ocaml packages are merged and released

**Comment by cerojasg1** [comment]
> yup this is still an issue. Should be fixed once some updates to some networking related ocaml packages are merged and released

Hi, Hasn't it been fixed yet?

**Comment by vbakke** [comment]
Hi @ajbt200128, do you have a rough guestimate on a timeline for this?
I'd like to introduce Semgrep to my company. But we are behind an https inspecting proxy. And I don't want to start off giving people a bad experience. (Bad selling point. : )

**Comment by ajbt200128** [comment]
Hey y'all, the PR we're waiting on is here https://github.com/mirage/ocaml-cohttp/pull/1080, so I would subscribe to that, as that is what we are waiting on

**Comment by cerojasg1** [comment]
Hi @ajbt200128 the PR you mentioned was merged, so will the issue be fix?

**Comment by vbakke** [comment]
Any progress on this @ajbt200128 ?
Or a rough timeline?

**Comment by ajbt200128** [comment]
we haven't had the bandwidth to fix this, but we should be able to get around to it in the next month or so. BTW we love contributions https://github.com/semgrep/semgrep if you want to take a crack at it yourself

**Comment by ajbt200128** [comment]
this should be fixed!

**Comment by cerojasg1** [comment]
> this should be fixed!

it's working, @ajbt200128  So I don't need install all phyton env anymore in windows. do I? Just install the extension and ready to go. Because I'm not sure if it's working because I had already installed semgrep CLI before
